lib/game_team.rb

```diff
@@ -20,29 +20,70 @@ def self.from_csv(csv_file_path)
   end
 
   def self.home_games
-    (@@all.find_all {|gt| gt.hoa == "home" }).count
+    (all.find_all {|gt| gt.hoa == "home" }).count
   end
 
   def self.percentage_home_wins
-    home_wins = (@@all.find_all {|gt| gt.hoa == "home" && gt.result == "WIN" }).count.to_f
+    home_wins = (all.find_all {|gt| gt.hoa == "home" && gt.result == "WIN" }).count.to_f
     ((home_wins / self.home_games) * 100).round(2)
   end
 
   def self.percentage_visitor_wins
-    visitor_wins = (@@all.find_all {|gt| gt.hoa == "home" && gt.result == "LOSS" }).count.to_f
+    visitor_wins = (all.find_all {|gt| gt.hoa == "home" && gt.result == "LOSS" }).count.to_f
     ((visitor_wins / self.home_games) * 100).round(2)
   end
 
   def self.percentage_ties
-    games_count = @@all.count.to_f
-    ties_count = (@@all.find_all { |gt| gt.result == "TIE"}).count.to_f
+    games_count = all.count.to_f
+    ties_count = (all.find_all { |gt| gt.result == "TIE"}).count.to_f
     ((ties_count / games_count) * 100).round(2)
   end
 
-  def self.winningest_coach
+  def self.coaches_in_season(season_id)
+    search_term = season_id.to_s[0..3]
+    game_teams_in_season = all.find_all do |gt|
+      gt.game_id.to_s[0..3] == search_term
+    end
+    game_teams_in_season.map {|gameteam| gameteam.head_coach}.uniq
+  end
+
+  def self.results_by_coach(season_id)
+    search_term = season_id.to_s[0..3]
+    results_by_coach_by_season = Hash.new { |hash, key| hash[key] = [] }
+    coaches_in_season(season_id).each do |coach|
+      all.find_all do |gt|
+        if gt.game_id.to_s[0..3] == search_term && coach == gt.head_coach
+            results_by_coach_by_season[coach] << gt.result
+        end
+      end
+    end
+    results_by_coach_by_season
+  end
+
```

Comment:
Believe it or not these were actually refactored to have less ends at the end!

Reply:
progress!

Reply:
We can refactor this to utilize find_by(season_id) which would delete out a few lines